src/Controllers/Content.php

```diff
@@ -80,6 +80,7 @@ public function index()
             $course_name = ParamValidate::getKeyInCustomFields($_SESSION, 'course_name');
 
             if (empty($custom_email_id) && isset($_SESSION['lti_post'][LTIConstants::LIS_PERSON_CONTACT_EMAIL_PRIMARY])) {
+                $course_id = $_SESSION['lti_post'][LTIConstants::CONTEXT_ID];
```

Comment:
1- This check is under the wrong if statement. Either create its own if{} block or handle it properly.
2- It's not just the course id that we need to override. We also need to look at the parameters we're overriding in the Content.php while deeplinking. We should override all those.